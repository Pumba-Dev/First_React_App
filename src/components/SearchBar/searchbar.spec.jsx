import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import {SearchBar} from "."

describe('<SearchBar />', () => {
    it('should have a value of search', () => {
        const fn = jest.fn();
        render(<SearchBar handleChange={fn} searchValue={'input test'} />);
        const $input = screen.getByPlaceholderText(/type to search/i);
        expect($input.value).toBe('input test');
    });

    it('should call handleChange function on each key pressed', () => {
        const fn = jest.fn();
        render(<SearchBar handleChange={fn}/>);
        const $input = screen.getByPlaceholderText(/type to search/i);
        const inputValue = 'input test';
        userEvent.type($input, inputValue);
        expect($input.value).toBe(inputValue);
        expect(fn).toHaveBeenCalledTimes(inputValue.length);
    });

    it('should match snapshot', () => {
        const fn = jest.fn();
        const {container} = render(<SearchBar handleChange={fn} searchValue={'input test'} />);
        expect(container.firstChild).toMatchSnapshot();
    });
});