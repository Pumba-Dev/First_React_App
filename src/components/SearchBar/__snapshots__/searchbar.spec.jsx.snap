// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar /> should match snapshot 1`] = `
<input
  class="search-bar"
  placeholder="Type to Search"
  type="search"
  value="input test"
/>
`;
